src/components/DrawingBoard.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { MouseEvent, useEffect, useRef } from 'react';
+import './DrawingBoard.css';
+
+export interface DrawingBoardProps {
+    imageUrl: string;
+    index: number;
+}
+
+enum drawingState {
+    NotStarted,
+    StartPoint,
+    Completed,
+}
+
+interface DrawingLine {
+    x1: number,
+    y1: number,
+    x2: number,
+    y2: number,
+}
+
+
+const DrawingBoard: React.FC<DrawingBoardProps> = (props: DrawingBoardProps) => {
+
+    let history: DrawingLine[] = [];
+
+    // save the reference to the canvas html element
+    const canvasRef = useRef<HTMLCanvasElement>(null);
+
+    let currentDrawing = {
```

Comment:
All of currentDrawing represents the state of the DrawingBoard component, so use useState in order to host it and render the component based on it:
NotStarted = empty
StartPoint = draw a point on the canvas
Completed =draw the line

In addition, the currentDrawing.state can be derived from the state (current saved dots) and managing it is redundant.

After you get a sense of the state, I suggest lifting it up to the Board component and treat the DrawingBoard as a stateless component. DrawingBoard is eventually the view layer which should be dumb as possible and receive the actual drawing as a prop.